-------------------------------------------------------------------------------
Test set: com.example.demo.services.UserServiceTest
-------------------------------------------------------------------------------
Tests run: 6, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.295 s <<< FAILURE! - in com.example.demo.services.UserServiceTest
shouldAddUser  Time elapsed: 0.065 s  <<< ERROR!
org.mockito.exceptions.misusing.WrongTypeOfReturnValue: 

UserDto cannot be returned by saveUser()
saveUser() should return User
***
If you're unsure why you're getting above error read on.
Due to the nature of the syntax above problem might occur because:
1. This exception *might* occur in wrongly written multi-threaded tests.
   Please refer to Mockito FAQ on limitations of concurrency testing.
2. A spy is stubbed using when(spy.foo()).then() syntax. It is safer to stub spies - 
   - with doReturn|Throw() family of methods. More in javadocs for Mockito.spy() method.

	at com.example.demo.services.UserServiceTest.shouldAddUser(UserServiceTest.java:114)

